assets/src/edit-story/components/canvas/gridview/index.js

```diff
@@ -73,21 +95,43 @@ const Rectangle = styled.div`
   }
 `;
 
-function RangeControl({ value, onChange }) {
+// I think we can move this component to shared one in the future if we need this kind of spacer span.
+const SpacerSpan = styled.span`
+  margin-left: ${({ space }) => space || 8}px;
+`;
+
+function RangeControl({ value, onChange, min, max, step }) {
+  const updateRangeValue = (addition) => {
+    onChange(Math.min(max, Math.max(min, value + addition)));
+  };
+
   return (
     <RangeInputWrapper>
-      <Rectangle>
+      <Rectangle
+        onClick={() => updateRangeValue(-step)}
+        disabled={value === min}
+        aria-label={__('Decrease thumbnail size', 'web-stories')}
+      >
         <RectangleIcon />
       </Rectangle>
-      <RangeInput
-        min="1"
-        max="3"
-        step="1"
+      <SpacerSpan space={20} />
+      <FlexGrowRangeInput
+        min={min}
+        max={max}
+        step={step}
         value={value}
         onChange={(evt) => onChange(Number(evt.target.value))}
         thumbSize={24}
+        aria-label={__('Thumbnail size', 'web-stories')}
+        aria-valuetext={__('Large', 'web-stories')}
```

Comment:
This should be dynamic depending on the `value`. If `value === min`, it should be `Small`, if `value === max` it should be `Large`. Otherwise `Medium` (we only have 3 steps here)

Reply:
Done